src/engine/controls/cuecontrol.cpp

```diff
@@ -554,30 +554,8 @@ void CueControl::reloadCuesFromTrack() {
     if (!m_pLoadedTrack)
         return;
 
-    // Determine current playing position of the track.
-    TrackAt trackAt = getTrackAt();
-    bool wasTrackAtZeroPos = isTrackAtZeroPos();
-    bool wasTrackAtIntroCue = isTrackAtIntroCue();
-
     // Update COs with cues from track.
     loadCuesFromTrack();
-
-    // Retrieve current position of cues from COs.
-    double cue = m_pCuePoint->get();
-    double intro = m_pIntroStartPosition->get();
-
-    // Make track follow the updated cues.
-    SeekOnLoadMode seekOnLoadMode = getSeekOnLoadPreference();
-
-    if (seekOnLoadMode == SeekOnLoadMode::MainCue) {
-        if ((trackAt == TrackAt::Cue || wasTrackAtZeroPos) && cue != Cue::kNoPosition) {
```

Comment:
I think we need here a check that verifies that the MainCue was at ZeroPos. 
We need to store the value before loadCuesFromTrack()

Reply:
That could still produce undesired seeking if the user has set the MainCue at 0:00 intentionally. Can we explicitly track whether this function is being called because the analyzer just finished?

Reply:
Only if you set the MainCue to a position away from the play-position.
This is currently not possible from the UI.

Reply:
I'd still be more comfortable with explicit tracking of what we want to track than indirectly deriving it from other information. There is less room for weird edge cases we are not thinking about to be affected and the code could be more straightforward by explicitly tracking whether this cue reload is coming from the analyzer.

Reply:
Thanks for your feedback and explaining the reasons behind the intentions of the behavior. I would also prefer an explicit solution, but if this introduces a lot of overhead I can also live with @daschuer 's suggestion to check if it was the main cue that was modified and if it was at 0:00 before.